net/xia/ppal_u6id/main.c

```diff
@@ -0,0 +1,241 @@
+#include <linux/init.h>
+#include <linux/in6.h>
+#include <linux/module.h>
+#include <net/ipv6.h>
+#include <net/ip6_checksum.h>
+#include <net/udp.h>
+#include <net/udp_tunnel.h>
+#include <net/xia_dag.h>
+#include <net/xia_list_fib.h>
+#include <net/xia_u6id.h>
+#include <net/xia_vxidty.h>
+#include <uapi/linux/udp.h>
+
+/* U6ID Principal */
+#define IPV6_ADDR_LEN 16;
+
+/* U6ID context */
+
+struct xip_u6id_ctx {
+	struct xip_ppal_ctx ctx;
+
+	struct socket __rcu *tunnel_sock;
+
+	/* Anchor for ill-formed U6ID XIDs. */
+	struct xip_dst_anchor ill_anchor;
```

Comment:
Where is ill_anchor being used in the code except when entries are added to it?

Reply:
It is used to store xid's that doesn't represent a well formed u6id xid i.e 16 bytes of ip address, 2 bytes port number and 2 bytes padded with zeroes. The anchor is freed in net_exit , before the module is unloaded.